pkg/deployment/deployment_item.go

```diff
@@ -85,11 +137,38 @@ func NewDeploymentItem(ctx SharedContext, project *DeploymentProject, collection
 		di.renderedYamlPath = filepath.Join(di.RenderedDir, ".rendered.yml")
 	}
 
-	err = di.Project.loadVarsList(di.VarsCtx, di.Config.Vars)
+	// update context vars with project/parrent path,references and tags
+	div := NewDeploymentItemVars(di.Project, di.Config, di.dir)
+
+	// temp
+	//k := di.Tags.ListKeys()
+	// i := len(k)
+	// for j, t := range k {
+	// 	i = i - 1
+	// 	div.Tags.Set(fmt.Sprintf("tag%d", j), t) // as kluctl auto-generated tag-N)
+	// }
+
+	pe := strings.Split(di.RelToSourceItemDir, "/")
+	pr := append([]string{}, pe...) // reversed
+	for i := len(pr)/2 - 1; i >= 0; i-- {
+		opp := len(pr) - 1 - i
+		pr[i], pr[opp] = pr[opp], pr[i]
+	}
+	div.RelPath = pe
+	div.RevPath = pr
+	div.RelRoot = strings.Repeat("../", len(strings.Split(di.RelToSourceItemDir, "/")))
```

Comment:
what does this try to achieve?

Reply:
it creates `../../../` ,, assume as prefix to get project root.

what ever you are nested, to load any shared component from project root is as easy as 

```
cluster/baseLayer/etcd/kustomization.yml
4:- {{ this.relRoot }}/compose/_dockerconfig
5:- {{ this.relRoot }}/compose/_wingmansigner
```
(compose ^^ is dir with shared components) that are re-used many times in various app deployments.

Reply:
Simply using `/compose/_dockerconfig` does not work? If I remember correct, it was at least supposed to work...if not, it's maybe a bug.

Reply:
For kustomize itself it would certainly not work if the structure is like below
```
├── compose
│   ├── dockerconfig
│                                     
├── cluster
│   ├── deployment.yml
│   ├── argo
│   │   ├── kustomization.yml
├── deployment.yml
├── .kluctl.yml
```


```
✗ Building kustomize objects
✗ 1 error occurred:
	* building kustomize objects for /Users/xxxx/Workspace/sre-staging-model/deployment/kube/cluster/argo failed. accumulating components: loader.New "new root '/compose/dockerconfig' cannot be absolute"
```


to make it working, you would have to on temp.  build dir update paths of `bases:, resources:, components:`

Reply:
Ok I see, even if kluctl templating would be able to load from `/` (pointing to the project root), kustomize would fail as it simply does not allow absolute pathes.

I'll think about this and maybe create one or two commits in regard to naming of these vars.